src/iqm/qiskit_iqm/fake_backends/iqm_fake_backend.py

```diff
@@ -292,15 +303,60 @@ def run(self, run_input: Union[QuantumCircuit, list[QuantumCircuit]], **options)
         Raises:
             ValueError: If empty list of circuits is provided.
         """
-        circuits = [run_input] if isinstance(run_input, QuantumCircuit) else run_input
+        circuits_aux = [run_input] if isinstance(run_input, (QuantumCircuit, IQMCircuit)) else run_input
 
-        if len(circuits) == 0:
+        if len(circuits_aux) == 0:
             raise ValueError("Empty list of circuits submitted for execution.")
 
+        class moves_to_swaps(TransformationPass):
+            """Checks for validity of move gates and changes them to swaps."""
+
+            def run(self, dag):
+                move_as_unitary = IQMCircuit(2)
+                move_as_unitary.unitary(MoveGate().unitary, [0, 1], label = "move")
+                move_as_unitary_dag = circuit_to_dag(move_as_unitary)
```

Comment:
we can remove this part and rely on transpile now that we have a unitary defined
```suggestion
```